crates/sui-core/src/authority.rs

```diff
@@ -2439,6 +2459,10 @@ impl AuthorityState {
         state
     }
 
+    pub fn get_cache_reader(&self) -> &Arc<ExecutionCache> {
```

Comment:
Should this return something about dyn ExecutionCacheReader?

Reply:
refactored to use traits everywhere. It was quite painful, lmk what you think